sirepo/auth/ldap.py

```diff
@@ -0,0 +1,56 @@
+"""LDAP login
+:copyright: Copyright (c) 2018-2022 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+from pykern import pkconfig
+from pykern.pkdebug import pkdlog, pkdp
+from pykern.pkcollections import PKDict
+import ldap
+import sirepo.quest
+
+AUTH_METHOD = "ldap"
+
+AUTH_METHOD_VISIBLE = True
+
+_cfg = None
+
+class API(sirepo.quest.API):
+    @sirepo.quest.Spec("require_cookie_sentinel")
+    def api_authLdapAuthorized(self):
+
+        # TODO(BG): pass simulation type as method parameter to assign to 'type', currently passing simulation_type to req and leaving 'type' False
```

Comment:
Simulation_type is fine in parse_post. Just rename the post value to be simulationType, and it will be parsed automatically.